hooks/src/internal.d.ts

```diff
@@ -52,12 +54,12 @@ export type HookState =
 
 interface BaseHookState {
 	_value?: unknown;
-	_nextValue?: undefined;
-	_pendingValue?: undefined;
-	_args?: undefined;
-	_pendingArgs?: undefined;
-	_component?: undefined;
-	_cleanup?: undefined;
+	_nextValue?: unknown;
+	_pendingValue?: unknown;
+	_args?: unknown;
+	_pendingArgs?: unknown;
+	_component?: unknown;
+	_cleanup?: unknown;
 }
```

Comment:
`undefined` is a more specific type than `unknown` which disallows some interfaces from extending these properties as they need to.